apps/backend/src/datasources/postgres/CallDataSource.ts

```diff
@@ -310,14 +310,24 @@ export default class PostgresCallDataSource implements CallDataSource {
               submission_message: args.submissionMessage,
               survey_comment: args.surveyComment,
               proposal_workflow_id: args.proposalWorkflowId,
-              call_ended:
-                preUpdateCall.call_ended &&
-                args.endCall &&
-                args.endCall.getTime() < currentDate.getTime(),
-              call_ended_internal: args.endCallInternal
-                ? preUpdateCall.call_ended_internal &&
-                  args.endCallInternal.getTime() < currentDate.getTime()
-                : args.callEndedInternal,
+              /*
+               If the call is updated with future end dates, set the ended flags as
+               false. If updated with past dates, leave the flags unchanged from the
+               existing values (if the existing values are false, the call end events
+               will fire for this call and update the flags, and if true it indicates
+               an old call being updated).
+              */
+              call_ended: args.callEnded
+                ? args.callEnded
+                : args.endCall && args.endCall.getTime() > currentDate.getTime()
+                ? false
+                : preUpdateCall.call_ended,
+              call_ended_internal: args.callEndedInternal
+                ? args.callEndedInternal
+                : args.endCallInternal &&
+                  args.endCallInternal.getTime() > currentDate.getTime()
+                ? false
+                : preUpdateCall.call_ended_internal,
```

Comment:
This logic looks like the same can we potential extract it out into sometime simple like 
if (args.callEndedInternal) {
    return preUpdateCall.call_ended_internal,
} else {

}

Reply:
I did originally have it extracted to a function, but in the end I thought it was a bit overkill and the ternary was more understandable, even if there is some duplication. I can change it if you think it's better.

```typescript
        /*
         Work out whether the call_ended and call_ended_internal flags need updating.
        */
        const determineCallEndedFlag = (
          newFlagValue: boolean | undefined,
          previousFlagValue: boolean | undefined,
          endCall: Date | undefined
        ) => {
          // Use the new value if explicitly passed in.
          if (newFlagValue) {
            return newFlagValue;
          }

          /*
           If the call end date has been changed to the future, set
           the flag to false.
          */
          if (endCall && endCall.getTime() > currentDate.getTime()) {
              return false;
          }

          /*
           Where the date has been changed to the past, leave the flag unchanged from
           its previous value. If it's false, the call end event will fire for this 
           call and update the flags, and if true it indicates an old call being updated).
          */
          return previousFlagValue;
        };
```

Reply:
This looks good easy to read than the chained ternary operators .